README.md

```diff
@@ -1 +1,71 @@
-# action-artifacts
\ No newline at end of file
+# Action Artifacts
+
+This repository contain a GitHub Action to interact with the [Artifacts] service.
+
+The [Artifacts] service is responsible of storing and distributing any sort of files that was created in a CI pipeline.
+
+## Method
+
+This action contain different method of interacting with Artifacts. You will find below each method with their description.
+
+### Setup
+
+Set outputs that relate to the artifacts attached to your build. See more about the outputs below.
+
+Example:
+
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Retrieve my artifacts name
+        id: artifacts
+        uses: scality/action-artifacts@v1
+        with:
+          method: setup
+          url: my.artifacts.url
+          user: ${{ secret.ARTIFACTS_USER }}
+          password: ${{ secret.ARTIFACTS_PASSWORD }}
+      - run: |
+          curl ${{ steps.artifacts.outputs.artifact-link }}/my-file -o my-file
+```
+
+### Upload
+
+Upload files to artifacts.
+
+### Promote
+
+Artifacts can be promoted when a tag has been made to make a build available forever.
+
+### Prolong
+
+Artifacts have an expiration date of 15 days. The prolong method allows you to make a copy of your build that will last longer.
+
+TODO: Add examples
+
+## Inputs
+
+This action take the following inputs:
+
+* `url`: The url of artifacts. This input is required.
+* `user`: The user to authenticate with Artifacts.
+Usually setup as a secret in the organization, ask your admin for the key of it. This input is required.
+* `password`: The password to authenticate the operation with Artifacts. This input is required.
+* `method`: What kind of interaction you will perform with Artifacts.
+With the choice of: `setup`, `upload`, `promote`, `prolong`.
+This input is required.
+* `sources`: File or directory to upload. To be used with the upload method.
+* `name`: The name of the artifacts build you will use. Only used with `prolong` and `promote` method.
+* `tag`: The git tag name of the artifacts you are going to promote. To be used with `promote` method.
+
+## Outputs
+
+This action will output the following variables:
+
+* `name`: Will contain a generated name for the artifacts that will be used for a workflow. Each workflow will have a unique artifact name which can be shared among the jobs that are within.
+* `link`: The full url in which artifacts will be stored.
```

Comment:
@erwan-b Have a look at the inputs and outputs. I took the liberty to change the name of some:
Input:
* artifacts-url -> set to url
* username -> user

Output:
* artifacts-name -> name
* artifacts-link -> link

I'm keeping them shorter to ease up the use and length of it. Per example on the output, I believe it make more sense if we do this:
```
# setting up artifacts like this:
- id: artifacts
  name: artifacts setup
  uses: scality/action-artifacts
  with:
    user: user
    url: url
    password: password
    method: setup
# allow me to call the output like this:
- name: echo
  run: echo ${{ steps.artifacts.outputs.name }} # So it sounds less repetitive than artifacts.output.artifacts-name
```